test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2603,6 +2672,110 @@ private Stream GetManifestStream(Stream packageStream)
             return resultStream;
         }
 
+        /// <summary>
+        /// Package directory for testing embedded icon functionality
+        /// </summary>
+        public sealed class IconTestSourceDirectory : IDisposable
```

Comment:
nested classes probably shouldn't be public.

Reply:
I made the class public because the other *SourceDirectory classes are also public. I will make this class protected to see how it works.